hack/update-deps.sh

```diff
@@ -46,21 +46,15 @@ drop-dep() {
   sed -i -e "\|//$1:go_default_library|d" "$path"
 }
 
-main() {
-  pushd "$(dirname "${BASH_SOURCE}")/.."
-  dep ensure
-  dep prune
-  hack/update-bazel.sh
-  drop-dep vendor/golang.org/x/text/language vendor/golang.org/x/text/internal
-  drop-dep vendor/google.golang.org/api/transport/grpc vendor/google.golang.org/api/transport
-  drop-dep vendor/github.com/golang/protobuf/protoc-gen-go/grpc vendor/github.com/golang/protobuf/protoc-gen-go
-  drop-dep vendor/github.com/golang/protobuf/protoc-gen-go/generator vendor/github.com/golang/protobuf/protoc-gen-go
-  hack/prune-libraries.sh --fix
-  hack/update-bazel.sh  # Update child :all-srcs in case parent was deleted
-}
-
-if ! main; then
-  echo FAILED >&2
-  exit 1
-fi
+trap 'echo "FAILED" >&2' ERR
```

Comment:
Will this still exit 1 on ERR?

Reply:
Yep, doesn't affect error code:

```
$ grep
Usage: grep [OPTION]... PATTERN [FILE]...
Try 'grep --help' for more information.
$ echo $?
2
$ cat /tmp/err.sh 
#!/bin/bash

trap 'echo "FAILED" >&2' ERR

grep
$ /tmp/err.sh 
Usage: grep [OPTION]... PATTERN [FILE]...
Try 'grep --help' for more information.
FAILED
$ echo $?
2
```

Reply:
Hmmm I guess I answered a different question -- do you need it to `exit 1` specifically on an error or do you want it to forward the error code that the script failed with?